core/src/main/scala/org/http4s/headers/Host.scala

```diff
@@ -17,14 +17,31 @@
 package org.http4s
 package headers
 
-import org.http4s.parser.HttpHeaderParser
+import cats.parse.Parser
+import org.http4s.internal.parsing.Rfc3986
 import org.http4s.util.Writer
+import scala.util.Try
 
 object Host extends HeaderKey.Internal[Host] with HeaderKey.Singleton {
   def apply(host: String, port: Int): Host = apply(host, Some(port))
 
   override def parse(s: String): ParseResult[Host] =
-    HttpHeaderParser.HOST(s)
+    ParseResult.fromParser(parser, "Invalid Host")(s)
+
+  //  // Do not accept scoped IPv6 addresses as they should not appear in the Host header,
+  //  // see also https://issues.apache.org/bugzilla/show_bug.cgi?id=35122 (WONTFIX in Apache 2 issue) and
+  //  // https://bugzilla.mozilla.org/show_bug.cgi?id=464162 (FIXED in mozilla)
```

Comment:
This was a rule in RFC2616 (which referred to the RFC2396 URI, and predates the current URI).  It's allowed in RFC7230.  I think those bugzillas issues date back to the older spec.

Reply:
just to be sure that I under stand RFC7230. we could just delegate to `Uri.Host.parser` now? and remove the comment

Reply:
https://github.com/http4s/http4s/pull/4208/commits/190d6766dc60f05351db722b8ae03ba4b15765de